controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/RunSerializer.java

```diff
@@ -102,10 +105,22 @@ Run runFromSlime(Slime slime) {
     }
 
     private Run runFromSlime(Inspector runObject) {
-        EnumMap<Step, Status> steps = new EnumMap<>(Step.class);
-        runObject.field(stepsField).traverse((ObjectTraverser) (step, status) -> {
-            steps.put(stepOf(step), stepStatusOf(status.asString()));
-        });
+        var steps = new EnumMap<Step, StepInfo>(Step.class);
+        runObject.field(steps2Field).traverse(((ObjectTraverser) (step, info) -> {
+            Inspector startTimeValue = info.field(startTimeField);
+            Optional<Instant> startTime = startTimeValue.valid() ?
+                    Optional.of(instantOf(startTimeValue.asLong())) :
+                    Optional.empty();
+            Step typedStep = stepOf(step);
+            steps.put(typedStep, new StepInfo(typedStep, stepStatusOf(info.field(statusField).asString()), startTime));
+        }));
+        if (steps.isEmpty()) {
```

Comment:
This saves you one release, but an upgraded controller will discard any changes made to the serialised data by an old controller. Three steps are needed to avoid this.

Reply:
Ignore this, as discussed.

Reply:
Changed to keep steps field to contain statuses, but add a new stepDetails field that contains startTimestamp and that may be extended with additional fields later.